src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -0,0 +1,302 @@
+      boolean requiredOverride = false;
+
+      if (parentClass != null) {
+        HaxeMethodModel method = parentClass.getMethod(((HaxeMethod)element).getModel().getName());
+        if (method != null) {
+          requiredOverride = true;
+        }
+      }
+
+      //System.out.println(aClass);
+      if (overrideAttribute != null && !requiredOverride) {
+        HaxeSemanticError.addError(
+          element,
+          new HaxeSemanticError(
+            "Overriding nothing",
+            new HaxeSemanticIntentionAction("Fix override") {
+              @Override
+              public void run() {
+                HaxePsiUtils.replaceElementWithText(overrideAttribute, "");
+              }
+            }
+          )
+        );
+      } else if (overrideAttribute == null && requiredOverride) {
+        HaxeSemanticError.addError(
+          element,
+          new HaxeSemanticError(
+            "Must override"
+          )
+        );
+      }
+      //PackageChecker.check((HaxePackageStatement)element);
+    }
+
+    for (ASTNode node : element.getNode().getChildren(null)) {
+      analyze(node.getPsi());
+    }
+  }
+}
+
+class PackageChecker {
+  static public void check(final HaxePackageStatement element) {
+    final HaxeReferenceExpression expression = ((HaxePackageStatement)element).getReferenceExpression();
+    String packageName = (expression != null) ? expression.getText() : "";
+    PsiDirectory fileDirectory = element.getContainingFile().getParent();
+    List<PsiFileSystemItem> fileRange = PsiFileUtils.getRange(PsiFileUtils.findRoot(fileDirectory), fileDirectory);
+    fileRange.remove(0);
+    String actualPath = PsiFileUtils.getListPath(fileRange);
+    final String actualPackage = actualPath.replace('/', '.');
+    final String actualPackage2 = HaxeResolveUtil.getPackageName(element.getContainingFile());
+    // @TODO: Should use HaxeResolveUtil
+
+    for (String s : StringUtils.split(packageName, '.')) {
+      if (!s.substring(0, 1).toLowerCase().equals(s.substring(0, 1))) {
+        HaxeSemanticError.addError(element, new HaxeSemanticError("Package name '" + s + "' must start with a lower case character"));
+      }
+    }
+
+    if (!packageName.equals(actualPackage)) {
+      HaxeSemanticError.addError(
+        element,
+        new HaxeSemanticError(
+          "Invalid package name! '" + packageName + "' should be '" + actualPackage + "'",
+          new HaxeSemanticIntentionAction("Fix package") {
+            @Override
+            public void run() {
+              Document document =
+                PsiDocumentManager.getInstance(element.getProject()).getDocument(element.getContainingFile());
+
+              if (expression != null) {
+                TextRange range = expression.getTextRange();
+                document.replaceString(range.getStartOffset(), range.getEndOffset(), actualPackage);
+              }
+              else {
+                int offset =
```

Comment:
I have checked it:

```
$ haxe Test.hx
Test.hx:2: lines 2-3 : Field a has less visibility (public/private) than superclass one
```
```
class Test extends Test2 {
        override private function a() {
        }
        static public function main() {
        }
}

class Test2 {
        public function a() {
        }
}
```

So I will change it to allow making it more visible but not the counterpart.